cmd/add_blob.go

```diff
@@ -22,14 +25,28 @@ func NewAddBlobCmd(blobsDir boshreldir.BlobsDir, fs boshsys.FileSystem, ui boshu
 }
 
 func (c AddBlobCmd) Run(opts AddBlobOpts) error {
-	file, err := c.fs.OpenFile(opts.Args.Path, os.O_RDONLY, 0)
-	if err != nil {
-		return bosherr.WrapErrorf(err, "Opening blob")
+	var file io.ReadCloser
+	var err error
+	href := ""
+	if u, err := url.ParseRequestURI(opts.Args.Path); err == nil && u.Scheme != "" && u.Host != "" {
+		resp, err := http.Get(opts.Args.Path)
+		if err != nil {
+			return bosherr.WrapErrorf(err, "Downloading blob")
+		}
+		defer resp.Body.Close() //nolint:errcheck
+		file = resp.Body
+		href = opts.Args.Path
+	} else {
+		file, err = c.fs.OpenFile(opts.Args.Path, os.O_RDONLY, 0)
+		if err != nil {
+			return bosherr.WrapErrorf(err, "Opening blob")
+		}
+		defer file.Close() //nolint:errcheck
 	}
 
 	defer file.Close() //nolint:errcheck
 
-	blob, err := c.blobsDir.TrackBlob(opts.Args.BlobsPath, file)
+	blob, err := c.blobsDir.TrackBlob(opts.Args.BlobsPath, file, href)
```

Comment:
Rather than handling the HREF -> `File` conversion here, I think it would make more sense either
-  `TrackBlob()` to take `opt.ArgsPath` itself, and handle writing the remote file into `tempFile` (https://github.com/cloudfoundry/bosh-cli/pull/683/files?diff=unified&w=1#diff-17144e6602ffb3d080986c8a0843436cd75cd11990789e6ba43d64ad98e9ed41L195)
- Have an explicit flag (ex: `--href` or `--uri`) that indicates that a remote file is being added
  - If the local file exists _and_ there is a `--uri` flag passed this would be stored for later verification

Reply:
If `TrackBlob()` is handling the fetching, perhaps a better interface would be to pass a `Blob{}` struct explicitly and allow that function to fill in `sha2` and `size`?